### ISSUE TITLE
Backends for different versions of gromacs

### ISSUE THREAD

**Issue by ajkluber** [ajkluber/model_builder#2]
With current developments we should consider refactoring the code that writes the .top file out of CoarseGrainedModel since writing the .top file may depend on what exact version of gromacs you are using. In particular, Fernando has begun using the SMOG2 gromacs which has a slightly different specification of pair interactions. Also we have a "using_sbm_gmx" config file option which is serving this purpose already (in a [hacky](https://github.com/ajkluber/model_builder/blob/master/models/CoarseGrainedModel.py#L374) way).

Basically this is just creating a logical frontend/backend separation. The frontend being the more abstract "model" which holds the interaction potentials that the user uses in parameter fitting for example. The backend being how the model is turned into files for simulation; the user doesn't care how it is done as long as it is compatible with the gromacs version. This should make things easier to maintain.

Thoughts?
